src/core/qfieldcloudconnection.cpp

```diff
@@ -580,21 +592,41 @@ int QFieldCloudConnection::uploadPendingAttachments()
         QgsMessageLog::logMessage( tr( "Failed to upload attachment stored at `%1`, reason:\n%2" )
                                      .arg( fileName )
                                      .arg( QFieldCloudConnection::errorString( attachmentReply ) ) );
+        mUploadFailingCount++;
       }
       else
       {
+        const int httpCode = attachmentReply->attribute( QNetworkRequest::HttpStatusCodeAttribute ).toInt();
+        if ( httpCode != 201 )
+        {
+          qInfo() << QStringLiteral( "Attachment project ID: %1" ).arg( projectId );
+          qInfo() << QStringLiteral( "Attachment file name: %1" ).arg( fileName );
+          qInfo() << QStringLiteral( "Attachment reply HTTP status code: %1" ).arg( httpCode );
+          for ( const QByteArray &header : attachmentReply->rawHeaderList() )
+          {
+            qInfo() << QStringLiteral( "Attachment reply header: %1 => %2" ).arg( header ).arg( attachmentReply->rawHeader( header ) );
+          }
+          qInfo() << QStringLiteral( "Attachment reply content: %1" ).arg( attachmentReply->readAll() );
+          AppInterface::instance()->sendLog( QStringLiteral( "QFieldCloud file upload HTTP code oddity!" ), QString() );
+        }
+
         QFieldCloudUtils::removePendingAttachment( projectId, fileName );
+        mUploadPendingCount--;
+        mUploadFailingCount = 0;
       }
 
-      mUploadCount--;
-      if ( mUploadCount == 0 )
+      if ( mUploadPendingCount > 0 && mUploadFailingCount < 5 )
+      {
+        // Move onto the next or retry the last attachment to upload
+        processPendingAttachments();
```

Comment:
Would be nice to launch this via a `QTimer::singleShot` with an exponentially increasing delay from let's say ...

Reply:
@m-kuhn , consider it done, really good idea.